src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -118,4 +116,50 @@ void AsyncOpenTask::unregister_download_progress_notifier(uint64_t token)
         m_session->unregister_progress_notifier(token);
 }
 
+void AsyncOpenTask::attach_to_subscription_initializer(AsyncOpenCallback&& async_open_callback,
+                                                       std::shared_ptr<_impl::RealmCoordinator> coordinator,
+                                                       bool rerun_subscription_on_open)
+{
+    // we need to wait until the initial subscription has been committed.
+    if (m_db_created || (rerun_subscription_on_open && !m_db_created)) {
```

Comment:
I think this check is not correct. Ideally it should be the same as when the callback was invoked (maybe there is a way to pass the result of that check)